compiler-core/src/language_server/engine.rs

```diff
@@ -55,6 +55,8 @@ pub struct LanguageServerEngine<IO, Reporter> {
     // Used to publish progress notifications to the client without waiting for
     // the usual request-response loop.
     progress_reporter: Reporter,
+
+    hex_deps: std::collections::HashSet<String>,
```

Comment:
Could you document this with a comment and use a `EcoString` here please? We don't use `String` other than with library that force it.